include/vctr/Containers/Span.h

```diff
@@ -103,31 +103,38 @@ class Span : public VctrBase<ElementType, std::span<ElementType, extent>, extent
         VCTR_ASSERT (extent == containerToView.size() || extent == std::dynamic_extent);
     }
 
+    /** Copies the data of the source container to the memory this span refers to. */
+    template <has::sizeAndDataWithElementType<ElementType> Container>
+    constexpr Span& operator= (const Container& containerToCopyDataFrom)
+    {
+        Vctr::copyFrom (containerToCopyDataFrom.data(), containerToCopyDataFrom.size());
+        return *this;
+    }
+
     /** Assigns the result of an expression to this vector.
 
         The Span has to match the size of the expression.
      */
     template <is::expression E>
     constexpr void operator= (const E& expression)
     {
-        if (! std::is_constant_evaluated())
-            VCTR_ASSERT (expression.size() == Vctr::size());
+        VCTR_ASSERT (expression.size() == Vctr::size());
```

Comment:
What makes this work now?

Reply:
I found out that since C++ 14 asserts actually work in constant expressions as long as the argument evaluates to true at compile time. I wasn't aware of this myself before. See https://akrzemi1.wordpress.com/2017/05/18/asserts-in-constexpr-functions/ for a bit more details